src/routes/blog/post/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/+page.markdoc

```diff
@@ -0,0 +1,340 @@
+
+For most databases, it’s beneficial for data to be structured. Structure gives data predictability and reliability, which is key to any production-ready application.
+
+SQL databases are also preferred when relations between data play a big role in preventing data duplication and ease of updating the data in one place. Postgres specifically allows you to add extensions to store and query geospatial and vector data efficiently.
+
+To integrate with Neon, you can set up an Appwrite function and prepare a Postgres client.
+
+```jsx
+import postgres from 'postgres';
+
+let client;
+
+export default async ({ req, res, log, error }) => {
+	// Connect to database
+  if(!client) {
+    const { PGHOST, PGDATABASE, PGUSER, PGPASSWORD, ENDPOINT_ID } = process.env;
+    client = postgres({
+      host: PGHOST,
+      database: PGDATABASE,
+      username: PGUSER,
+      password: PGPASSWORD,
+      port: 5432,
+      ssl: 'require',
+      connection: {
+        options: `project=${ENDPOINT_ID}`,
+      },
+    });
+  }
+
+  // Rest of the code
+};
+```
+
+Next, you can run SQL queries to create a table, insert data, read it, and much more.
+
+```jsx
+// Prepare table
+await client`CREATE TABLE IF NOT EXISTS warehouses (
+  id SERIAL PRIMARY KEY,
+  location VARCHAR(255),
+  capacity INT
+)`;
+
+// Insert data
+const location = `Street ${Math.round(Math.random() * 1000)}, Earth`; // Random address
+const capacity = 10 + Math.round(Math.random() * 10) * 10; // Random number: 10,20,30,...,90,100
+await client`INSERT INTO warehouses (location, capacity) VALUES (${location}, ${capacity})`;
+
+// Query data
+const page = 1;
+const limit = 100;
+const warehouses = await client`SELECT * FROM warehouses LIMIT ${limit} OFFSET ${(page - 1) * limit}`;
+return res.json(warehouses);
+```
+
+You can find an entire example in our [query Neon Postgres](https://github.com/appwrite/templates/tree/main/node/query-neon-postgres) function template, which is also available on your Appwrite Console in the Functions Marketplace.
+
+![Neon Postgres Console](/images/blog/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/neon.png)
+
+## MongoDB Atlas
+
+Atlas, a managed cloud platform by the MongoDB team, is a cloud provider of MongoDB clusters.
+
+MongoDB database is well-known for its JSON document structure. This approach allows you to store unstructured data optimally, with a huge improvement boost to write operations. For instance, usage stats or sales history could be stored in MongoDB and later processed to gain more insights into app analytics.
+
+While MongoDB shines the most in storing unstructured data, nowadays, it provides all the capabilities developers expect from a relational database. If you don’t like SQL queries, MongoDB is a great choice.
+
+With Mongo, integration becomes simpler, since you don’t need to define structure of collections or documents. After preparing a client, you can directly start writing and querying data.
```

Comment:
```suggestion
With MongoDB, integration becomes simpler, since you don’t need to define structure of collections or documents. After preparing a client, you can directly start writing and querying data.
```